eng/Directory.Build.Common.props

```diff
@@ -1,4 +1,4 @@
-﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" TreatAsLocalProperty="DefineConstants">
```

Comment:
Can you give more info on why you need this? I don't like adding these in the common props because it will block folks adding additional ones in their projects.

Reply:
If you pass `/p:DefineConstants=SNIPPET` it becomes a global variable that cannot be overridden without this, so any project that tries to add to `DefineConstants` will no-op.

Alternatively, I could add some other condition and arbitrary property to instead use an in-props `DefineConstants` if you'd prefer, but all this does is allow global variables to be overridden. IMO, passing constants via command line can be useful to conditionalize builds so there's little to no harm in this.

Reply:
...which is what I see you suggested below, so I'll just do that instead. Still, I think DefineConstants is one of those that would be safe, but I don't feel strongly about it.